models/spring-ai-azure-openai/src/main/java/org/springframework/ai/azure/openai/AzureOpenAiChatClient.java

```diff
@@ -256,4 +214,121 @@ private <T> List<T> nullSafeList(List<T> list) {
 		return list != null ? list : Collections.emptyList();
 	}
 
+	// JSON merge doesn't due to Azure OpenAI service bug:
+	// https://github.com/Azure/azure-sdk-for-java/issues/38183
+	private ChatCompletionsOptions merge(ChatCompletionsOptions azureOptions, AzureOpenAiChatOptions springAiOptions) {
+
+		if (springAiOptions == null) {
+			return azureOptions;
+		}
+
+		ChatCompletionsOptions mergedAzureOptions = new ChatCompletionsOptions(azureOptions.getMessages());
+		mergedAzureOptions.setStream(azureOptions.isStream());
+
+		mergedAzureOptions.setMaxTokens(azureOptions.getMaxTokens());
+		if (mergedAzureOptions.getMaxTokens() == null) {
+			mergedAzureOptions.setMaxTokens(springAiOptions.getMaxTokens());
+		}
+
+		mergedAzureOptions.setLogitBias(azureOptions.getLogitBias());
+		if (mergedAzureOptions.getLogitBias() == null) {
+			mergedAzureOptions.setLogitBias(springAiOptions.getLogitBias());
+		}
+
+		mergedAzureOptions.setStop(azureOptions.getStop());
+		if (mergedAzureOptions.getStop() == null) {
+			mergedAzureOptions.setStop(springAiOptions.getStop());
+		}
+
+		mergedAzureOptions.setTemperature(azureOptions.getTemperature());
+		if (mergedAzureOptions.getTemperature() == null && springAiOptions.getTemperature() != null) {
+			mergedAzureOptions.setTemperature(springAiOptions.getTemperature().doubleValue());
+		}
```

Comment:
Why have `if (mergedAzureOptions.getTemperature() == null && springAiOptions.getTemperature() != null) {` 

instead of `if (mergedAzureOptions.getTemperature() == null` as in the other cases?

Also we don't need to call `doubleValue()` in `mergedAzureOptions.setTemperature(springAiOptions.getTemperature().doubleValue());` as it accepts the wrapper number type directly.